### PR TITLE
Switch to hipBLASCommon

### PR DESCRIPTION
**Pull request by bethune-bryant**
This PR is now ready for reviews!

In this PR we remove the dependency on hipBLAS and switch to using the hipBLAS-common package.

### CODE REVIEW

**Review by bnemanich — DISMISSED** [review]

**Review by KKyang — DISMISSED** [review]
Please use squash merge, thanks!

**Review by jichangjichang — CHANGES_REQUESTED** [review]
Have you confirmed that rocm mainline build already has hipblas-common package installed?

**Review by jichangjichang — DISMISSED** [review]
lgtm, please use squash merge.
It is better to update https://github.com/ROCm/hipBLASLt/blob/develop/README.md as well.

File: library/CMakeLists.txt [hunk 1]
```diff
@@ -142,6 +142,7 @@ target_include_directories(hipblaslt
                                    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
                                    $<BUILD_INTERFACE:${HIP_INCLUDE_DIRS}>
                                    $<BUILD_INTERFACE:${HIPBLAS_INCLUDE_DIRS}>
+                                   $<BUILD_INTERFACE:${HIPBLAS-COMMON_INCLUDE_DIRS}>
```
Comment (AlexBrownAMD) [inline library/CMakeLists.txt]: Is HIPBLAS_INCLUDE_DIRS still required if this is using hipblas-common instead?

Comment (bnemanich) [reply library/CMakeLists.txt]: Oh, that is a good point. This should not be needed anymore. @awhittle3 can you please remove this?

Comment (awhittle3) [reply library/CMakeLists.txt]: Reference removed, builds as expected. I did not find any other references in the CMake files.

**Review by jichangjichang — APPROVED** [review]
lgtm, please use squash merge

**Review by bnemanich — APPROVED** [review]

**Pull request merged**